### PR TITLE
Naively running 2to3 over the codebase

### PR DESCRIPTION
**Pull request by dammitjim**
Also bumps version to 2.0.19

### CODE REVIEW

**Review by SemVerTsar — APPROVED** [review]

File: src/gdata/apps/groups/client.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 __author__ = 'Shraddha gupta <shraddhag@google.com>'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/apps/groups/client.py]: probably don't need anything other than `urllib.parse` here

File: src/gdata/apps/groups/service.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 __author__ = 'google-apps-apis@googlegroups.com'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/apps/groups/service.py]: here too

File: src/gdata/apps/multidomain/client.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 __author__ = 'Claudio Cherubino <ccherubino@google.com>'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/apps/multidomain/client.py]: here

File: src/gdata/apps/organization/client.py [hunk 4]
```diff
@@ -27,7 +27,7 @@
 __author__ = 'Gunjan Sharma <gunjansharma@google.com>'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/apps/organization/client.py]: here

File: src/gdata/apps/service.py [hunk 5]
```diff
@@ -25,7 +25,7 @@
       from xml.etree import ElementTree
     except ImportError:
       from elementtree import ElementTree
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/apps/service.py]: where are these being used in this file? Can't see where any lines using them have changed...

File: src/gdata/auth.py [hunk 6]
```diff
@@ -21,7 +21,7 @@
 import re
 import time
 import types
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/auth.py]: parse

File: src/gdata/calendar/client.py [hunk 7]
```diff
@@ -28,7 +28,7 @@
 __author__ = 'alainv (Alain Vongsouvanh)'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/calendar/client.py]: being used?

File: src/gdata/calendar/service.py [hunk 8]
```diff
@@ -15,19 +15,19 @@
...
+                         URL arguments (represented as strings). This is a
                          utility function used in CRUD operations.
 """
 
 
 __author__ = 'api.vli (Vivian Li)'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/calendar/service.py]: think only parse needed here

File: src/gdata/calendar_resource/client.py [hunk 9]
```diff
@@ -28,7 +28,7 @@
 
 import gdata.calendar_resource.data
 import gdata.client
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/calendar_resource/client.py]: parse

File: src/gdata/contentforshopping/client.py [hunk 10]
```diff
@@ -21,7 +21,7 @@
 __author__ = 'afshar (Ali Afshar), dhermes (Daniel Hermes)'
 
 
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/contentforshopping/client.py]: parse

File: src/gdata/docs/client.py [hunk 11]
```diff
@@ -21,7 +21,7 @@
 import copy
 import mimetypes
 import re
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/docs/client.py]: parse

File: src/gdata/docs/service.py [hunk 12]
```diff
@@ -32,7 +32,7 @@
 import atom
 import gdata.service
 import gdata.docs
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/docs/service.py]: parse

File: src/gdata/gauth.py [hunk 13]
```diff
@@ -49,8 +49,8 @@
 import datetime
 import time
 import random
-import urllib
-import urlparse
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/gauth.py]: duplicate import, only parse needed

File: src/gdata/marketplace/client.py [hunk 14]
```diff
@@ -26,7 +26,7 @@
 
 import gdata.marketplace.data
 import gdata.client
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (allthatilk) [inline src/gdata/marketplace/client.py]: parse

**Comment by dammitjim** [comment]
@allthatilk I've modified the places where 2to3 was being a bit overenthusiastic here: https://github.com/omni-digital/gdata-python-client/pull/1/commits/d93b7f16d1ab79fde164dfa1e3bbb86bd2e19f9d

File: src/gdata/marketplace/client.py [hunk 15]
```diff
@@ -26,11 +26,11 @@
 
 import gdata.marketplace.data
 import gdata.client
-import urllib
+import urllib.request, urllib.parse,
```
Comment (allthatilk) [inline src/gdata/marketplace/client.py]: parse

File: src/gdata/oauth/__init__.py [hunk 16]
```diff
@@ -1,529 +1,529 @@
...
-        # concatenate the consumer key and secret
-        sig = escape(consumer.secret) + '&'
-        if token:
-            sig = sig + escape(token.secret)
-        return sig
-
-    def build_signature(self, oauth_request, consumer, token):
-        return self.build_signature_base_string(oauth_request, consumer, token)
+import cgi
+import urllib.parse
```
Comment (allthatilk) [inline src/gdata/oauth/__init__.py]: duplicate import

File: src/gdata/sample_util.py [hunk 17]
```diff
@@ -21,7 +21,7 @@
 
 import sys
 import getpass
-import urllib
+import urllib.parse
```
Comment (allthatilk) [inline src/gdata/sample_util.py]: is this one being used?

**Review by allthatilk — CHANGES_REQUESTED** [review]
A few small tweaks left

File: src/atom/http_interface.py [hunk 18]
```diff
@@ -20,20 +20,23 @@
...
 
 
 __author__ = 'api.jscudder (Jeff Scudder)'
 
 
-import StringIO
+try:
+    from io import StringIO
+except ImportError:
+    from io import StringIO
```
Comment (dammitjim) [inline src/atom/http_interface.py]: This doesn't need a try / except

**Review by allthatilk — APPROVED** [review]

**Comment by allthatilk** [comment]
LGTM

**Pull request merged**